cmd/daemon.go

```diff
@@ -54,8 +54,7 @@ var daemonCmd = &cobra.Command{
 		case "stop":
 			err = s.Stop()
 		case "run":
-			m := daemon.NewDaemon()
-			m.Serve()
+			err = s.Run(daemon.NewDaemon().Run(context.Background()))
```

Comment:
We should check if the error is nil or not. If not, then we need to log it out.

Reply:
Also, this line is a little bit unintuitive to read. Could we do a little bit of separation here?

```
m := daemon.NewDaemon()
onStart, onStop := m.Run(context.Background())
err = s.Run(onStart, onStop)
...
```